HAR-4102 Ylläpidon Sanktioraportti

jarnovayrynen opened the pull request.

jarnovayrynen committed:
Luo ylläpitoluokalle koodisto

Luo ylläpitoluokalle koodisto
Käytä tiemerkinän muut työt -toteumalomakkeella ylläpitoluokan koodistoa

jarnovayrynen committed:
Paranna dokumentaatiota ylläpitoluokkien osalta

jarnovayrynen committed:
Mahdollista sanktioraportti myös ylläpidon urakoissa

Vaadi kantatasolla että ylläpitoluokka on kokonaisluku väliltä 1-10.

jarnovayrynen committed:
Pompauta migraationumeroa

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HAR-4102

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HAR-4102

Kauko committed:
Lisää raporteille :arvo-ja-yksikko solu

jarnovayrynen committed:
Käytä ylläpitoluokka domainia eri näkymistä

Käytä ylläpitoluokka domainia kohdeluettelosta, tiemerkinnän yks.hint. töistä
ja tiemerkinnän muista töistä.

Lisää datamuunnin yllapitokohteet-domain/yllapitoluokka-xf

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HAR-4102

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HAR-4102

jarnovayrynen committed:
Pompauta migraationumeroa

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HAR-4102

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HAR-4102

jarnovayrynen committed:
Lisää ylläpidon sanktioraportti

Lisää uusi raportti Ylläpidon sanktioraportti

Validoi ylläpitoluokka oltava kokonaisluku väliltä 1-10 (YHA-skeema)

Aseta sanktiolle teoreettinen max määrä.

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HAR-4102

jarnovayrynen committed:
Lisää ylläpidon sanktioraportille testit

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HAR-4102

jarnovayrynen committed:
Tarjoa DNS:ttyä testiympäristöä Hipchatissä

jarnovayrynen committed:
Merge branch 'develop' into HAR-4102

jarnovayrynen committed:
Merge branch 'develop' into HAR-4102

jarnovayrynen commented on tietokanta/src/main/resources/db/migration/V1_499__.sql:
```diff
@@ -0,0 +1,15 @@
+
+ALTER TABLE sanktio
+  ADD CONSTRAINT sakoille_maxmaara
+CHECK (maara < 1000000000); -- yli 1000Me:n sakkoja tuskin tulee
```
Tämä vastaava validoidaan myös frontissa nyt. Jos ennen tätä constraintia syötti "ihan hirveän ison luvun", saattoi saada selaimeen ihan mitä sattuu numeroita. Siksi constraint.

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HAR-4102

Kauko commented on src/cljs/harja/views/urakka/laadunseuranta/laatupoikkeama.cljs:
```diff
@@ -160,7 +160,7 @@ sekä sanktio-virheet atomin, jonne yksittäisen sanktion virheet kirjoitetaan (
             :nimi        :summa
             :leveys      1
-            :validoi     [[:ei-tyhja "Anna sakon summa euroina"]]
+            :validoi     [[:ei-tyhja "Anna sakon summa euroina"] [:rajattu-numero nil 0 999999999 "Anna arvo väliltä 0 - 999 999 999"]]
```
Mitä jos joku kusee hommat niin pahasti, että saa miljardin euron edestä sakkoa? :^)

jarnovayrynen replied:
Sitten tulee bugi. :)

Kauko commented on src/clj/harja/palvelin/raportointi/raportit/sanktiot.sql:
```diff
@@ -34,4 +34,45 @@ WHERE ((:urakka::INTEGER IS NULL AND u.urakkanro IS NOT NULL) OR u.id = :urakka)
+                                                                                              :hallintayksikko) AND u.urakkanro IS NOT NULL))
+      AND (:urakka::INTEGER IS NOT NULL OR
+           (:urakka::INTEGER IS NULL AND (:urakkatyyppi :: urakkatyyppi IS NULL OR
+                                          u.tyyppi = :urakkatyyppi :: urakkatyyppi)))
```
Näyttää duplikaattiehdoilta nyt ylemmän :urakka tarkastuksen kanssa

Kauko commented on src/clj/harja/palvelin/raportointi/raportit/sanktioraportti_yllapito.clj:
```diff
@@ -0,0 +1,186 @@
+(ns harja.palvelin.raportointi.raportit.sanktioraportti-yllapito
```
Namespace dokumentaation vois lisätä.

Musta ainakin tuntuu, että kannattais dokumentoida, mitä sarakkeita raportilla kuuluu olla. Voi vanheta, mutta vanhentuneestakin tiedosta saa kiinni, että mikä raportin pointti on.

jarnovayrynen replied:
Hyvä kommentti, lisäsin dokumentaation.

jarnovayrynen committed:
Eriytä sanktioraporttien yhteinen koodi

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HAR-4102

jarnovayrynen committed:
Merge branch 'HAR-4102' of github.com:finnishtransportagency/harja into HAR-4102

jarnovayrynen committed:
Poista ylläpitourakan sanktioraportista indeksit

jarnovayrynen committed:
HAR-4660 Lisää Aikataulunäkymään YP-lk muotoilija

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HAR-4102

jarnovayrynen committed:
Tarkenna Ylläpidon Sakko- ja bonusraporttia

Poista yksiköt rivin otsikosta koska jo summassa fmt:n avulla
Järjestä SQL-kyselyt niin että urakkaehdot ennen hallintayksikköehtoja.

jarnovayrynen committed:
Korjaa Sanktioraportin testit vastaamaan muutoksia

jarnovayrynen committed:
Kutsu ylläpidon täbiä Sakot ja bonukset

jarnovayrynen committed:
Näytä sanktiot-välilehdelläkin yp-sakot negatiivisena

Ylläpidon Sakko- ja bonusraportti näyttää sakot miinusmerkkisenä. Lisää
yhdenmukaisuuden vuoksi sama logiikka Ylläpidon Sakot ja bonukset -näkymään.

jarnovayrynen committed:
Lisää ns-dokumentaatio Ylläpidon Sanktioraporttiin

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HAR-4102

jarnovayrynen committed:
Lisää urakkatason Sanktiorapparitestiin urakkatyyppi

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HAR-4102

jarnovayrynen committed:
Pompauta migraationumeroa

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HAR-4102

Kauko committed:
Merge branch 'develop' into HAR-4102

Kauko committed:
Yksittäinen raporttielementti voi asettaa formatoinnin pois

Tarve ilmeni sanktioraportista. Yhdessä sarakkeessa on muistutuksia (kpl),
ja sakkoja (euro). Muistutukset piirrettiin [:arvo-ja-yksikko] elementtinä,
joissa yksikkönä kpl, mutta koska sarakkeen :fmt oli :raha, piirrettiin
yksiköksi kpl JA euro. Nyt solu asettaa :fmt? false, jolloin formatter
ei koske siihen.

Kauko committed:
Sanktioraportin muistutus on [:arvo-ja-yksikko] elementti

Asetetaan solun yksiköksi kpl, ja ohitetaan solun kohdalla formatter,
joka on solun sarakkeessa :raha.

Kauko committed:
Poista välilyönti arvo-ja-yksikko raporttielementistä

Kauko committed:
Korjaa sanktioraportin testi, muistutus on raporttisolu
